### DISCUSSION TITLE
How should we format Regex in allowedPostUpgradeCommands?

### DISCUSSION THREAD

**Discussion by jpgoelz** [renovatebot/renovate#25693] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.53.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

How do we need to format the Regex in `allowedPostUpgradeCommands`?

I saw that the Regex for `allowedVersions` ([docs](https://docs.renovatebot.com/configuration-options/#allowedversions)) uses double escapes like we use in Java, e.g. `\\w` instead of `\w`.

Is this also true for `allowedPostUpgradeCommands`?

That would explain, why `^cd [\w-]+(\/[\w-]+)*$` did not seem to work before in [my last question](https://github.com/renovatebot/renovate/discussions/25607).

**Comment by rarkins** [comment]
It depends on how/where you're configuring it, e.g. in JS or JSON. JSON needs additional escaping

**Comment by jpgoelz** [comment]
We are using config.js.

**Comment by jpgoelz** [comment]
So, we tried using JS-Regex-Objects, but they do not seem to be picked up correctly. They appear as [Object object] in our error message:
![image](https://github.com/renovatebot/renovate/assets/32487587/c142b900-4c89-46ba-8aa4-e0291b034181)

This is really odd, because this message does not appear, when using strings. And I checked the input against the regEx() function in regex.ts in the renovate Project using tests.

I am at a loss, because I do not have access to the logs and don't understand why there would be an error.

**Reply by jpgoelz** [reply, marked as answer]
I did my own inquiry into the matter:
Discussion: #25792 
PR: #25793
